### ISSUE TITLE
aws_ecs_cluster rename causes crash

### ISSUE THREAD

**Issue by airhorns** [hashicorp/terraform#2427]
Not exactly sure what I did to cause the crash, and I'm not too good with go stack traces, but I've managed to get terraform into a state where it crashes every time after trying to rename an aws_ecs_cluster resource. I believe that the inconsistent state may have been caused by the fact that ECS clusters can't be destroyed or recreated if there are active services within it (which I didn't know until the destroy failed), after which I went and destroyed all the services manually in the UI. 

```
$ terraform --version
Terraform v0.6.0-dev (7850bb39d5465be88ef05e1efb7067788e5cf094)
```

Log is here: https://www.dropbox.com/s/ahmxsj18p9y7o54/crash.log?dl=0

**Comment by radeksimko** [comment]
Hi @airhorns 
thanks for reporting this edge case.

I tried renaming an ECS cluster and have not seen the crash, yet I have seen the following error:
```
* ClusterContainsServicesException: The Cluster cannot be deleted while Services are active.
```

Either way, both should be easy to fix. If you really do decide to rename the cluster (recreate everything all over again), Terraform could try and wait until all active ECS services are actually dead (i.e. no `ClusterContainsServicesException`) and keep trying a few minutes.

To your crash - that should be also fairly easy to fix - probably just need to check length of returned services.

**Note to myself:** I should not reference `[0]` index if I don't know for sure there will be one -> this applies to any resource.

**Comment by radeksimko** [comment]
It is not possible to move services between clusters (unique service consists of service name+cluster), hence the service needs to be recreated in the new cluster all over again -> `ForceNew: true` was missing, that's also why you were hitting the crash.

See #2698 which should address all the issues mentioned here.

**Comment by radeksimko** [comment]
Fixed via #2698
